artichoke-backend/src/extn/core/symbol/mod.rs

```diff
@@ -1,115 +1,14 @@
 use std::ffi::c_void;
 
 use crate::convert::{Immediate, UnboxedValueGuard};
-use crate::extn::core::array::Array;
 use crate::extn::prelude::*;
-use crate::intern::Symbol as SymbolId;
 
 pub mod ffi;
 pub mod mruby;
 pub mod trampoline;
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Symbol(SymbolId);
```

Comment:
there were 3 different `Symbol` wrappers around `u32` floating around in `artichoke-backend`:

- this one.
- `Symbol` in `intern` module.
- `intaglio::Symbol`.

One of the nice simplifications of this PR is replacing all of the Ruby Core pieces with `spinoso_symbol::Symbol` and replacing the rest of the internals with `u32`, only marshaling to the `intaglio` repr at the edge.